src/lib/Components/Bidding/Screens/ConfirmBid.tsx

```diff
@@ -30,12 +33,122 @@ interface ConfirmBidProps extends ViewProperties {
     cents: number
   }
+  relay?: RelayPaginationProp
+  navigator?: NavigatorIOS
+}
+
+interface ConformBidState {
+  pollCount: number
+  intervalToken: number
 }
 
-export class ConfirmBid extends React.Component<ConfirmBidProps> {
+const MAX_POLL_ATTEMPTS = 20
+
+const bidderPositionMutation = graphql`
+  mutation ConfirmBidMutation($input: BidderPositionInput!) {
+    createBidderPosition(input: $input) {
+      result {
+        position {
+          id
+        }
+        status
+        message_header
+        message_description
+        message_description_md
+      }
+    }
+  }
+`
+
+export class ConfirmBid extends React.Component<ConfirmBidProps, ConformBidState> {
+  state = {
+    pollCount: 0,
+    intervalToken: 0,
+  }
+
   onPressConditionsOfSale = () => {
     SwitchBoard.presentModalViewController(this, "/conditions-of-sale?present_modally=true")
   }
+
+  placeBid() {
+    commitMutation(this.props.relay.environment, {
+      onCompleted: (results, errors) => {
+        this.verifyBidPosition(results, errors)
+      },
+      onError: e => {
+        // TODO catch error!
+        // this.verifyAndShowBidResult(null, e)
+        console.log("error!", e, e.message)
+      },
+      mutation: bidderPositionMutation,
+      variables: {
+        input: {
+          sale_id: this.props.sale_artwork.sale.id,
+          artwork_id: this.props.sale_artwork.artwork.id,
+          max_bid_amount_cents: this.props.bid.cents,
+        },
+      },
+    })
+  }
+
+  verifyBidPosition(results, errors) {
+    const status = results.createBidderPosition.result.status
+    if (!errors && status === "SUCCESS") {
+      const positionId = results.createBidderPosition.result.position.id
+      const query = `
+        {
+          me {
+            bidder_position(id: "${positionId}") {
+              processed_at
+              is_active
+            }
+          }
+        }
+      `
+      const interval = setInterval(() => {
+        metaphysics({ query }).then(this.checkBidPosition.bind(this))
+      }, 1000)
+      this.setState({ intervalToken: interval })
+    } else {
+      const message_header = results.createBidderPosition.result.message_header
+      const message_description_md = results.createBidderPosition.result.message_description_md
+      this.showBidResult(false, message_header, message_description_md)
+    }
+  }
+
+  checkBidPosition(result) {
+    // TODO: move polling logic to a separate file https://github.com/artsy/emission/pull/1025#discussion_r185931915
+    const bidderPosition = result.data.me.bidder_position
+    if (bidderPosition.processed_at) {
+      clearInterval(this.state.intervalToken)
+      if (bidderPosition.is_active) {
+        // wining
+        this.showBidResult(true)
+      } else {
+        // outbid
+        this.showBidResult(false)
+      }
+    } else {
+      // poll again
+      if (this.state.pollCount > MAX_POLL_ATTEMPTS) {
+        clearInterval(this.state.intervalToken)
```

Comment:
Can we add a TODO here to insert the "Bid result not available" screen if we've passed the max number of attempts? I'm not sure if that's implied by this.